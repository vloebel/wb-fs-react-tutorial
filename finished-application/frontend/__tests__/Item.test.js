import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import ItemComponent from '../components/Item';
import { fakeItem } from '../lib/testUtils';

const item = fakeItem();

const mocks = {};

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const { container } = render(
      <MockedProvider>
        <ItemComponent item={item} />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    render(
      <MockedProvider>
        <ItemComponent item={item} />
      </MockedProvider>
    );
    const img = screen.getByAltText(item.name);
    expect(img).toBeInTheDocument();
  });

  it('renders the pricetag and title', () => {
    const { container } = render(
      <MockedProvider>
        <ItemComponent item={item} />
      </MockedProvider>
    );
    expect(screen.getByText('$50')).toBeInTheDocument();
    const link = container.querySelector('a');
    expect(link).toHaveAttribute('href', '/product/abc123');
    expect(link).toHaveTextContent(item.name);
  });

  it('renders out the buttons properly', () => {
    const { container } = render(
      <MockedProvider>
        <ItemComponent item={item} />
      </MockedProvider>
    );

    const edit = screen.getByText(/Edit/i);
    expect(edit.href).toContain('update?id=abc123');

    const addToCart = screen.getByText(/add to cart/i);
    expect(addToCart).toHaveProperty('type', 'button');
    expect(addToCart).toBeInTheDocument();

    const deleteItem = screen.getByText(/delete/i);
    expect(deleteItem).toHaveProperty('type', 'button');
    expect(deleteItem).toBeInTheDocument();
  });
});
