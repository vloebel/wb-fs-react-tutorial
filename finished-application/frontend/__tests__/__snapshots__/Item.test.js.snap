// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<div>
  <div
    class="ItemStyles__Item-sc-16pk14u-0 biTBO"
  >
    <img
      alt="dogs are best"
      src="dog.jpg"
    />
    <h3
      class="Title-sc-16nq74k-0 cPJKCc"
    >
      <a
        href="/product/abc123"
      >
        dogs are best
      </a>
    </h3>
    <span
      class="PriceTag-nwbk6t-0 ffeJgy"
    >
      $50
    </span>
    <p>
      dogs
    </p>
    <div
      class="buttonList"
    >
      <a
        href="/update?id=abc123"
      >
        Edit ✏️
      </a>
      <button
        type="button"
      >
        Add
         To Cart 🛒
      </button>
      <button
        type="button"
      >
        Delete This Item
      </button>
    </div>
  </div>
</div>
`;
